tests/integration/test_permissions.py

```diff
@@ -0,0 +1,337 @@
+        password=encrypted_password,
+        active=True
+    )
+    user_partially_allowed = User(
+        email='partially_allowed@inspirehep.net',
+        password=encrypted_password,
+        active=True,
+    )
+    user_allowed = User(
+        email='allowed@inspirehep.net',
+        password=encrypted_password,
+        active=True,
+    )
+
+    db.session.add_all(
+        [user, user_partially_allowed, user_allowed]
+    )
+    db.session.commit()
+
+    # Create actions for the allowed user
+    restricted_collection_action = ActionUsers(
+        action='view-restricted-collection',
+        argument='Restricted Collection',
+        user_id=user_allowed.id
+    )
+    another_restricted_collection_action = ActionUsers(
+        action='view-restricted-collection',
+        argument='Another Restricted Collection',
+        user_id=user_allowed.id
+    )
+
+    # Create actions for the partially allowed user
+    partial_restricted_collection_action = ActionUsers(
+        action='view-restricted-collection',
+        argument='Restricted Collection',
+        user_id=user_partially_allowed.id
+    )
+
+    db.session.add_all(
+        [
+            restricted_collection_action,
+            another_restricted_collection_action,
+            partial_restricted_collection_action
+        ]
+    )
+    db.session.commit()
+
+    yield
+
+    SessionActivity.query.delete()
+    ActionUsers.query.filter_by(action='view-restricted-collection').delete()
+    User.query.filter_by(email='user@inspirehep.net').delete()
+    User.query.filter_by(email='partially_allowed@inspirehep.net').delete()
+    User.query.filter_by(email='allowed@inspirehep.net').delete()
+    db.session.commit()
+
+
+def test_all_collections_are_cached(app, users, sample_record):
+    """Test that collection info gets cached."""
+    with app.test_client() as client:
+        # Remove collection cache key
+        cache.delete('restricted_collections')
+
+        client.get("/literature/123")
+
+        # Check that cache key has been correctly filled
+        assert pickle.loads(cache.get('restricted_collections')) == \
+            set([u'Another Restricted Collection', u'Restricted Collection'])
+
+
+@pytest.mark.parametrize('user_info,status', [
+    # anonymous user
+    (None, 200),
+    # Logged in user without permissions assigned
+    (dict(email='user@inspirehep.net'), 200),
+    # Logged in user with permissions assigned
+    (dict(email='allowed@inspirehep.net'), 200),
+    # admin user
+    (dict(email='admin@inspirehep.net'), 200),
+])
+def test_record_public_detailed_read(app, sample_record, users, user_info, status):
+    """Test that a record can be read by everybody through web interface."""
+    with app.test_client() as client:
+        if user_info:
+            # Login as user
+            login_user_via_session(client, email=user_info['email'])
+
+        result = client.get("/literature/123")
+        assert result.status_code == status
```

Comment:
Since you are not going to use `result` further IMO `assert client.get('/literature/123').status_code == status` is enough.